doc/source/whatsnew/v1.4.4.rst

```diff
@@ -22,6 +22,7 @@ Fixed regressions
 - Fixed regression in :meth:`DataFrame.loc` not updating the cache correctly after values were set (:issue:`47867`)
 - Fixed regression in :meth:`DataFrame.loc` not aligning index in some cases when setting a :class:`DataFrame` (:issue:`47578`)
 - Fixed regression in :meth:`DataFrame.loc` setting a length-1 array like value to a single value in the DataFrame (:issue:`46268`)
+- Fixed regression when slicing with :meth:`DataFrame.loc` with :class:`DateOffset`-index (:issue:`46671`)
```

Comment:
DateOffset-index is unclear.  maybe "DatetimeIndex with a pd.DateOffset object for its ``freq``"?

Reply:
This is already merged. Shall I file another PR to improve the message?

Reply:
included in #48314